lib/linalg/Vector.h

```diff
@@ -759,6 +759,27 @@ class Vector
      */
     double localMin(int nmax = 0);
 
+    /**
+     * @brief Distribute this vector among MPI processes,
+     * based on the specified local dimension.
+     * This becomes distributed after this function is executed.
+     *
+     * @pre !distributed()
+     * @pre d_owns_data
+     *
+     * @param[in] local_dim dimension for local MPI rank.
+     */
+    void distribute(const int &local_dim);
```

Comment:
These functions would be more general if an MPI communicator is an argument, with default value `MPI_COMM_WORLD`.

Reply:
On second thought, `MPI_COMM_WORLD` is used in other places in the class, so the `Vector` constructor should take an optional argument for the communicator. It's not really necessary to do this in this PR, but it's a possibility.

Reply:
It was also one of my wish, that libROM parallel linear algebra should support custom MPI communicator. I attempted it a while ago, but it wasn't as simple as I thought, particularly in the interface with scalapack. We can discuss about this more.

For now, in order to be consistent with the current practice throughout the code, I kept `MPI_COMM_WORLD`.